### PR TITLE
[Torch] Fix conv2d conversion for group conv (group > 1 but != in channels)

### PR DESCRIPTION
**Pull request by masahi**
`groups` > 1 case is hard coded to do depth wise conv, for group conv (groups != in_channels) it raises a shape error.

please review @icemelon9 @anijain2305 @kevinthesun @jwfromm @alexwong

### CODE REVIEW

File: python/tvm/relay/frontend/pytorch.py [hunk 1]
```diff
@@ -250,7 +250,9 @@ def _impl(inputs, input_types):
         channels = weight_shape[0]
         groups = int(inputs[8])
 
-        if groups > 1:
+        # weight_shape[1] is always in_channels // groups
+        # For depthwise, in_channels == groups
```
Comment (jwfromm) [inline python/tvm/relay/frontend/pytorch.py]: This comment is a little confusing, Make it clear that we're checking if this is a depthwise conv and that if weight_shape[1] != 1, then its a group conv.

Comment (masahi) [reply python/tvm/relay/frontend/pytorch.py]: Added more comments. Also explained why we need reshape for depth wise case.

**Comment by masahi** [comment]
ready to merge @icemelon9 @anijain2305 @kevinthesun @tqchen @yzhliu

**Comment by tqchen** [comment]
Thanks @masahi @jwfromm !

**Pull request merged**